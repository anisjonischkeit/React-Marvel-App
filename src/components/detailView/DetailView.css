.detailViewCard {
	height: 100vh;
	overflow-y: scroll;
	-webkit-overflow-scrolling: touch;
}

@media only screen and (max-width: 960px) {
	.detailViewCard {
		height: calc(100vh - 64px);
	}
}
