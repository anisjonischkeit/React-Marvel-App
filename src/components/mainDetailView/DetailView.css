.detailView {
	/* width: 100%; */
	height: 100vh;
	overflow-y: auto;
	margin: 0;
	width: calc(100% - 400px);
	float: left;
}