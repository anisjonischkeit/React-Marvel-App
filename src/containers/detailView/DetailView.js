import * as React from 'react';
import DetailViewComponent from '../../components/detailView/DetailView';
import DetailItemProperties from '../../components/detailView/DetailItemProperties';

import type { DataType } from '../../reducers/data'

import { bindActionCreators } from 'redux';
import { fetchIndividualDataItem, selectDataItem } from '../../actions/data/'
import { withRouter } from 'react-router-dom'

import { CardText } from 'material-ui/Card';
import { connect } from 'react-redux';

import Detail from '../../components/masterDetailView/Detail'

type DataObjItemType = {
	id: number,
	thumbnail: {
		path: string,
		extension: string
	},
	description: string,
	characters: {
		items: Array<*>
	},
	comics: {
		items: Array<*>
	}
}

type PropsType = {
	selectedId : number,
	dataObj : {
		[id : number] : DataObjItemType
	},
	firstItem : DataObjItemType,
	history : { push : (url: string) => void },
	statNames : Array<string>,
	displayName : string,
	handleBackClick : () => void,
	titleFieldName : string,
	fetchIndividualDataItem : (dataName: string, resourceURI: string) => void 
}

export const DetailViewContainer = (props : PropsType) => {
	
	const onClickHandler = (dataName, resourceURI) => {
		props.fetchIndividualDataItem(dataName, resourceURI)
		props.history.push(`/${dataName}`)
	}

 	if (props.selectedId != null && props.dataObj) {
		let selected = props.dataObj[props.selectedId];

		if (selected == null) {
			if (props.firstItem && props.selectedId === props.firstItem.id) {
				selected = props.firstItem
			} else {
				return null
			}
		}
		
		let itemProperties = props.statNames.map(statName => ({
				...selected[statName],
				onClickHandler: onClickHandler.bind(null, statName),
				name: statName
			})
		)

		return (
			<Detail
				title={props.displayName}
				onBackClick={props.handleBackClick}
			>
				<DetailViewComponent
					title={selected[props.titleFieldName]}
					subtitle={String(selected.id)}
					img={`${selected.thumbnail.path}/landscape_incredible.${selected.thumbnail.extension}`}
				>
					<div>
						<CardText>
							{selected.description}
						</CardText>
						
						<DetailItemProperties
							itemProperties={itemProperties}
						/>

					</div>
				</DetailViewComponent>
			</Detail>
		)
	}
	return null
}

const mapStateToProps = (state, props) => {
	return {
		selectedId: state.data[props.reduxEntryPointName].selectedId,
		dataObj: state.data[props.reduxEntryPointName].obj,
		firstItem: state.data[props.reduxEntryPointName].firstItem
	}
}

const mapDispatchToProps = (dispatch, props) => ({
  ...bindActionCreators({
		fetchIndividualDataItem,
		handleBackClick: () => selectDataItem(props.reduxEntryPointName, null)
  }, dispatch)
});



export default connect(mapStateToProps, mapDispatchToProps)(withRouter(DetailViewContainer))