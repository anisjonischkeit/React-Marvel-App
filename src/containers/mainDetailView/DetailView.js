import * as React from 'react';
import DetailViewComponent from 'components/mainDetailView/DetailView';
import DetailViewStats from 'components/mainDetailView/DetailViewStats';

import { bindActionCreators } from 'redux';
import { fetchIndividualDataItem, selectDataItem } from 'actions/data/'
import { withRouter } from 'react-router-dom'

import { CardText } from 'material-ui/Card';
import { connect } from 'react-redux';

import Detail from 'components/masterDetail/Detail'

const DetailViewContainer = withRouter(props => {
	
	const onClickHandler = (dataName, resourceURI) => {
		props.fetchIndividualDataItem(dataName, resourceURI)
		props.history.push(`/${dataName}`)
	}

 	if (props.selectedId && props.dataObj) {
		let selected = props.dataObj[props.selectedId];
		if (selected == null) {
			if (props.firstItem && props.selectedId === props.firstItem.id) {
				selected = props.firstItem
			} else {
				return null
			}
		}
		
		let stats = {}
		props.statNames.forEach(key => { 
			stats[key] = {...selected[key], onClickHandler: onClickHandler.bind(null, key)}
		})

		const descriptionWithStats = (
			<div>
				<CardText>
				 	{selected.description}
				</CardText>
				
				<DetailViewStats
					stats={stats}
				/>

			</div>
		)

		return (
			<Detail
				title={props.displayName}
				onBackClick={props.handleBackClick}
			>
				<DetailViewComponent
					title={selected[props.titleFieldName]}
					subtitle={selected.id}
					img={`${selected.thumbnail.path}/landscape_incredible.${selected.thumbnail.extension}`}
					description={descriptionWithStats}
				/>
			</Detail>
		)
	}
	return null
})

const mapStateToProps = (state, props) => {
	return {
		selectedId: state.data[props.reduxEntryPointName].selectedId,
		dataObj: state.data[props.reduxEntryPointName].obj,
		firstItem: state.data[props.reduxEntryPointName].firstItem
	}
}

const mapDispatchToProps = (dispatch, props) => ({
  ...bindActionCreators({
		fetchIndividualDataItem,
		handleBackClick: () => selectDataItem(props.reduxEntryPointName, null)
  }, dispatch)
});



export default connect(mapStateToProps, mapDispatchToProps)(DetailViewContainer)