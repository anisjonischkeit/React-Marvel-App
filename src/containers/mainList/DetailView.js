import * as React from 'react';
import DetailViewComponent from 'components/mainList/DetailView';

// import { bindActionCreators } from 'redux';
// import { fetchCharacters } from 'actions/data/characterActions'

import { connect } from 'react-redux';

const DetailViewContainer = props => {
	if (props.selectedId) {
		const selected = props.charactersObj[props.selectedId];
		return (
			<DetailViewComponent
				title={selected.name}
				subtitle={selected.id}
				img={`${selected.thumbnail.path}standard_small${selected.thumbnail.extension}`}
				description={selected.description}
			/>
		)
	}
	return null
}

const mapStateToProps = state => {
	return {
		selectedId: state.data.characters.selectedId,
		charactersObj: state.data.characters.obj
	}
}

export default connect(mapStateToProps)(DetailViewContainer)