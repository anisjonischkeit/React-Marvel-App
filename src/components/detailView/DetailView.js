import React from 'react';

import './DetailView.css';

import { Card, CardMedia, CardTitle } from 'material-ui/Card';

type PropsType = {
  title: string,
  subtitle: string,
  img: string,
  children: React$Node,
}

export default (props : PropsType) => (
  <Card className='detailViewCard'>
    <CardMedia
      overlay={<CardTitle title={props.title} subtitle={props.subtitle} />}
    >
      <img src={props.img} alt={props.title} />
    </CardMedia>
    {props.children}
  </Card>
)