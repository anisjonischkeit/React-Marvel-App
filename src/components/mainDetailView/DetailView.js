import React from 'react';

import './DetailView.css';

import { Card, CardActions, CardHeader, CardMedia, CardTitle, CardText } from 'material-ui/Card';
import FlatButton from 'material-ui/FlatButton';

export default (props) => (
	<Card className='detailView' >

		<CardMedia
			overlay={<CardTitle title={props.title} subtitle={props.subtitle} />}
		>
			<img src={props.img} alt={props.title} />
		</CardMedia>
		
		{props.description}
	</Card>
)