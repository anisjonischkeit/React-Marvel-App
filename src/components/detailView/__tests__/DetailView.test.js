import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json'

import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import DetailView from "../DetailView"

describe('DetailView', () => {
  describe("Renders correctly", () => {
    const itemProperties = mount(
      (<MuiThemeProvider>
				<DetailView
					title="Sample Title"
					subtitle="Sample SubTitle"
					img="http://sampleImgUrl.com"
				>
					<p id="sampleChild">sampleChild</p>
					<p id="sampleChild2">sampleChild</p>
				</DetailView>
      </MuiThemeProvider>)
    );

		it("Matches Snapshot", () => {
			let tree = toJson(itemProperties);
			expect(tree).toMatchSnapshot();
		})

		it("Renders Children", () => {
			expect(itemProperties.find("#sampleChild").exists()).toBeTruthy()
			expect(itemProperties.find("#sampleChild2").exists()).toBeTruthy()
			expect(itemProperties.find("#sampleChild1000").exists()).toBeFalsy()
		})
	})
	
})