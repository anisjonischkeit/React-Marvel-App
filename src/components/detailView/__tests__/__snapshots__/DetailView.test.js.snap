// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailView Renders correctly Matches Snapshot 1`] = `
<MuiThemeProvider>
  <Component>
    <Card
      className="detailViewCard"
      expandable={false}
      expanded={null}
      initiallyExpanded={false}
    >
      <Paper
        circle={false}
        className="detailViewCard"
        rounded={true}
        style={
          Object {
            "zIndex": 1,
          }
        }
        transitionEnabled={true}
        zDepth={1}
      >
        <div
          className="detailViewCard"
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0,0,0,0)",
              "backgroundColor": "#ffffff",
              "borderRadius": 2,
              "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                     0 1px 4px rgba(0, 0, 0, 0.12)",
              "boxSizing": "border-box",
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "Roboto, sans-serif",
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "zIndex": 1,
            }
          }
        >
          <div
            style={
              Object {
                "paddingBottom": 0,
              }
            }
          >
            <CardMedia
              key=".0"
              overlay={
                <CardTitle
                  subtitle={undefined}
                  title={undefined}
                />
              }
            >
              <div
                style={
                  Object {
                    "muiPrepared": true,
                    "position": "relative",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "muiPrepared": true,
                    }
                  }
                >
                  <img
                    key=".0"
                    style={
                      Object {
                        "maxWidth": "100%",
                        "minWidth": "100%",
                        "muiPrepared": true,
                        "verticalAlign": "top",
                        "width": "100%",
                      }
                    }
                  />
                </div>
                <div
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "muiPrepared": true,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "height": "100%",
                        "muiPrepared": true,
                        "position": "relative",
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "background": "rgba(0, 0, 0, 0.54)",
                          "bottom": 0,
                          "left": 0,
                          "muiPrepared": true,
                          "paddingTop": 8,
                          "position": "absolute",
                          "right": 0,
                        }
                      }
                    >
                      <CardTitle
                        key=".0"
                        subtitleColor="rgba(255, 255, 255, 0.54)"
                        titleColor="rgba(255, 255, 255, 0.87)"
                      >
                        <div
                          style={
                            Object {
                              "muiPrepared": true,
                              "padding": 16,
                              "position": "relative",
                            }
                          }
                        >
                          <span
                            style={
                              Object {
                                "color": "rgba(255, 255, 255, 0.87)",
                                "display": "block",
                                "fontSize": 24,
                                "lineHeight": "36px",
                                "muiPrepared": true,
                              }
                            }
                          />
                          <span
                            style={
                              Object {
                                "color": "rgba(255, 255, 255, 0.54)",
                                "display": "block",
                                "fontSize": 14,
                                "muiPrepared": true,
                              }
                            }
                          />
                        </div>
                      </CardTitle>
                    </div>
                  </div>
                </div>
              </div>
            </CardMedia>
            <p
              id="sampleChild"
              key=".1:0"
            >
              sampleChild
            </p>
            <p
              id="sampleChild2"
              key=".1:1"
            >
              sampleChild
            </p>
          </div>
        </div>
      </Paper>
    </Card>
  </Component>
</MuiThemeProvider>
`;
